source/text/Format.cpp

```diff
@@ -44,6 +47,91 @@ namespace {
 		reverse(result.begin(), result.end());
 	}
 
+	void ScanForSubstitutions(const string &source,
+			function<void(const string &, size_t, size_t)> ProcessText,
+			function<const string *(const string &)> SubstitutionFor)
+	{
+		string key;
+		size_t start = 0;
+		size_t search = start;
+		while(search < source.length())
+		{
+			size_t left = source.find('<', search);
+			if(left == string::npos)
+				break;
+
+			size_t right = source.find('>', left);
+			if(right == string::npos)
+				break;
+
+			++right;
+			size_t length = right - left;
+			key.assign(source, left, length);
+			const string *subs = SubstitutionFor(key);
+			if(subs)
+			{
+				ProcessText(source, start, left - start);
+				ProcessText(*subs, 0, string::npos);
```

Comment:
If we're using a abbreviation, use the singular `sub`.
`subs` could just be the first four characters of `substitution` but it looks more like the plural of `sub`, but this is only a single substitution.
```suggestion
			const string *sub = SubstitutionFor(key);
			if(sub)
			{
				ProcessText(source, start, left - start);
				ProcessText(*sub, 0, string::npos);
```

Reply:
I applied this suggestion  manually.